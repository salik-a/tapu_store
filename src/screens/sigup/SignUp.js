import React, { useState } from "react";
import { View, Text, SafeAreaView } from "react-native";
import styles from './SignUpStyle'
import Input from "../../components/input/Input"
import Button from "../../components/button/Button"
import { Formik } from "formik";
import auth from "@react-native-firebase/auth"


const Sign = ({ navigation }) => {

    const initialForm = {
        usermail: "",
        password: "",
        passwordRepeat: ""
    }
    const handleSign = async (formValues) => {
        if (formValues.password !== formValues.passwordRepeat) {

        } else {

            try {
                await auth().createUserWithEmailAndPassword(formValues.usermail, formValues.password);
                console.log("formValues", formValues);

            } catch (error) {

                console.log("error", error)
            }
        }
    }

    return (
        <SafeAreaView style={styles.container}>
            <Formik
                initialValues={initialForm}
                onSubmit={values => handleSign(values)}
            >

                {({ handleChange, handleSubmit, values }) => (
                    <>
                        <Input
                            placeholder="E-Postanızı Giriniz"
                            onChangeText={handleChange('usermail')}
                            value={values.usermail}
                        />
                        <Input
                            placeholder="Şifrenizi Giriniz"
                            onChangeText={handleChange('password')}
                            value={values.password}
                            isSecure={true}
                        />
                        <Input
                            placeholder="Şifrenizi  Tekrar Giriniz"
                            onChangeText={handleChange('passwordRepeat')}
                            value={values.passwordRepeat}
                            isSecure={true}
                        />
                        <View style={styles.buttons}>
                            <Button title="Kayıt Ol" bgColor='papayawhip' onPress={handleSubmit} />
                            <Button title="Geri" bgColor='papayawhip' onPress={() => navigation.goBack()} />
                        </View>
                    </>
                )

                }
            </Formik>

        </SafeAreaView>
    );
};

export default Sign;
