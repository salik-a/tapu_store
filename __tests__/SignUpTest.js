jest.mock('@react-native-firebase/auth', () => {
    return () => ({
        signInWithEmailAndPassword: jest.fn(),
        createUserWithEmailAndPassword: jest.fn(),
        signOut: jest.fn(),

    });
});
import React from 'react'
import { fireEvent, render, waitFor } from '@testing-library/react-native'
import SignUp from '../src/screens/sigup/SignUp';



test('should match with snapshot', () => {

    const comp = render(<SignUp />);
    expect(comp).toMatchSnapshot();
});

