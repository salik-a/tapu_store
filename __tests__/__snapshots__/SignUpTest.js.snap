// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "marginBottom": 15,
          "marginTop": 25,
          "width": "85%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 30,
            "fontWeight": "800",
            "marginVertical": 10,
          },
        ]
      }
    >
      Account
    </Text>
  </View>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "borderBottomWidth": 0.8,
            "borderColor": "#BBC3CF",
            "height": 50,
            "marginVertical": 20,
            "width": "85%",
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="E-Mail"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingBottom": 5,
              "paddingLeft": 5,
              "paddingRight": 5,
              "paddingTop": 5,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "borderBottomWidth": 0.8,
            "borderColor": "#BBC3CF",
            "height": 50,
            "marginVertical": 20,
            "width": "85%",
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="Password"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingBottom": 5,
              "paddingLeft": 5,
              "paddingRight": 5,
              "paddingTop": 5,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "borderBottomWidth": 0.8,
            "borderColor": "#BBC3CF",
            "height": 50,
            "marginVertical": 20,
            "width": "85%",
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="Username"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingBottom": 5,
              "paddingLeft": 5,
              "paddingRight": 5,
              "paddingTop": 5,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "borderBottomWidth": 2,
            "borderColor": "#64738C",
            "width": "85%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "width": "100%",
            },
            Object {
              "zIndex": 5000,
            },
            Object {},
            false,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onLayout={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "white",
              "borderColor": "white",
              "borderRadius": 8,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "space-between",
              "marginTop": 10,
              "opacity": 1,
              "paddingHorizontal": 10,
              "width": "100%",
              "zIndex": 5000,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "flex": 1,
                },
                Object {},
                Object {
                  "fontSize": 16,
                },
                false,
              ]
            }
          >
            Türkçe
          </Text>
          <View
            style={
              Array [
                Object {
                  "marginLeft": 10,
                },
                Object {},
              ]
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../react-native-dropdown-picker/src/themes/light/icons/arrow-down.png",
                }
              }
              style={
                Array [
                  Object {
                    "height": 20,
                    "width": 20,
                  },
                  Object {},
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 25,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#BBC3CF",
            "borderColor": "rgba(0, 0, 0, .0)",
            "borderRadius": 12,
            "borderWidth": 1,
            "height": 50,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
            "width": "85%",
          }
        }
        testID="button-touchable"
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 14,
                "textAlign": "center",
              },
            ]
          }
          testID="button-title"
          textColor="white"
        >
          Sign Up
        </Text>
      </View>
    </View>
  </View>
</View>
`;
